src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Localization;
+using Microsoft.Framework.WebEncoders;
+
+
+namespace Microsoft.AspNet.Mvc.Localization
+{
+    /// <summary>
+    /// An <see cref="IHtmlLocalizer"/> that uses the <see cref="IStringLocalizer"/> to provide localized HTML content.
+    /// This service just encodes the arguments but not the resource string.
+    /// </summary>
+    public class HtmlLocalizer : IHtmlLocalizer
+    {
+        private IStringLocalizer _localizer;
+        private IStringLocalizerFactory _localizerFactory;
+        private readonly IHtmlEncoder _encoder;
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlLocalizer"/>.
+        /// </summary>
+        /// <param name="factory">The <see cref="IStringLocalizerFactory"/>.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/>.</param>
+        public HtmlLocalizer([NotNull] IStringLocalizerFactory factory, [NotNull] IHtmlEncoder encoder)
+        {
+            _localizerFactory = factory;
+            _encoder = encoder;
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlLocalizer"/>.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IStringLocalizer"/> to read strings from.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/>.</param>
+        public HtmlLocalizer([NotNull] IStringLocalizer localizer, [NotNull] IHtmlEncoder encoder)
+        {
+            _localizer = localizer;
+            _encoder = encoder;
+        }
+
+        private IStringLocalizer Localizer
```

Comment:
Regression from @DamianEdwards 's design.

If we have this pattern elsewhere we should fix that. This seems like an anti-pattern to me. Getters should never throw, and parameter validation on private/internal properties is not useful to users because there is no immediate action they can take to them.